crm_salesperson_planner_reassign/models/crm_salesperson_planner_visit.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2022 Sygel - Manuel Regidor
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+from odoo import fields, models
+
+
+class CrmSalespersonPlannerVisit(models.Model):
+    _inherit = "crm.salesperson.planner.visit"
+
+    last_user_id = fields.Many2one(comodel_name="res.users", string="Last Salesperson",)
+
+    def write(self, vals):
+        if (
+            not self.env.context.get("bypass_update_event")
+            and vals.get("user_id")
+            and self.user_id
```

Comment:
This is not multi-record compliant and will give a singleton error in such case.